index.js

```diff
@@ -214,6 +224,75 @@ class Conf {
 		this.store = store;
 	}
 
+	_serializeExtraTypes(value) {
+		// Base cases
+		for (const type of this.extraTypes) {
+			if (value instanceof type.instance) {
+				return {
+					[TYPE_KEY]: type.name,
+					val: type.convertFrom(value)
+				};
+			}
+		}
+
+		// Sub-object handling of extra types
+		const stack = [];
+		stack.push(value);
+		while (stack.length > 0) {
```

Comment:
Why are you using a while loop?

Reply:
So in case of nested objects I want to reuse the same logic to serialize `Date` types, so there are two ways I can go about this. The first one is to use recursion, the other one is to use the iterative solution, which is the while loop.  
I've decided to use the iterative solution over the recursive solution, because with really deeply nested objects, there's a possibility that the function could overflow the callstack.